mne/io/reference.py

```diff
@@ -4,20 +4,51 @@
 #
 # License: BSD (3-clause)
 
+from copy import deepcopy
 import numpy as np
 
 from .constants import FIFF
 from .proj import _has_eeg_average_ref_proj, make_eeg_average_ref_proj
-from .pick import pick_types
+from .proj import setup_proj
+from .pick import pick_types, pick_channels
 from .base import BaseRaw
 from ..evoked import Evoked
 from ..epochs import BaseEpochs
 from ..utils import logger, warn, verbose
 
 
+def _copy_channel(inst, ch_name, new_ch_name):
+    """Add a copy of a channel specified by ch_name.
+
+    Input data can be in the form of Raw, Epochs or Evoked.
+
+    The instance object is modified inplace.
+
+    Parameters
+    ----------
+    inst : instance of Raw | Epochs | Evoked
+        Data containing the EEG channels
+    ch_name : str
+        Name of the channel to copy.
+    new_ch_name : str
+        Name given to the copy of the channel.
+
+    Returns
+    -------
+    inst : instance of Raw | Epochs | Evoked
+        The data with a copy of a given channel.
+    """
+    new_inst = inst.copy().pick_channels([ch_name])
+    new_inst.rename_channels({ch_name: new_ch_name})
+    inst.add_channels([new_inst], force_update_info=True)
+    return inst
```

Comment:
This function creates 2 copies of the data and is called later in `_set_bipolar_reference` in a for loop. Since setting a reference is typically done on the raw data, this may become a performance bottleneck.